cordapp-cpk/src/test/kotlin/net/corda/plugins/cpb/CpbTest.kt

```diff
@@ -0,0 +1,138 @@
+package net.corda.plugins.cpb
+
+import net.corda.plugins.cpk.GradleProject
+import net.corda.plugins.cpk.cordaApiVersion
+import net.corda.plugins.cpk.deleteAll
+import net.corda.plugins.cpk.expectedCordappContractVersion
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestReporter
+import org.junit.jupiter.api.io.TempDir
+import java.io.InputStream
+import java.nio.file.Files
+import java.nio.file.Path
+import java.security.DigestInputStream
+import java.security.MessageDigest
+import java.security.cert.X509Certificate
+import java.util.TreeMap
+import java.util.jar.JarInputStream
+import javax.security.auth.x500.X500Principal
+import kotlin.streams.asSequence
+
+class CpbTest {
+    companion object {
+        private const val cordappVersion = "1.2.1"
+        private const val hostVersion = "2.0.1"
+        private val cordaDevCertPrincipal = X500Principal("CN=Corda Dev Code Signer, OU=R3, O=Corda, L=London, C=GB")
+
+
+        private lateinit var externalProject: GradleProject
+        private lateinit var testProject: GradleProject
+        private lateinit var testDir: Path
+        private lateinit var mavenRepoDir : Path
+        private lateinit var externalCordappProjectDir : Path
+        private lateinit var cpbProjectDir : Path
+
+        @Suppress("unused")
+        @BeforeAll
+        @JvmStatic
+        fun setup(@TempDir tmpDir: Path, reporter: TestReporter) {
+            testDir = tmpDir
+            mavenRepoDir = Files.createDirectory(testDir.resolve("maven"))
+            externalCordappProjectDir = Files.createTempDirectory(testDir.parent, "externalCordapp")
+            cpbProjectDir = Files.createDirectory(testDir.resolve("cpb"))
```

Comment:
```kotlin
fun setup(@TempDir testDir: Path, reporter: TestReporter) {
    val mavenRepoDir = Files.createDirectory(testDir.resolve("maven"))
    val externalCordappProjectDir = Files.createTempDirectory(testDir.parent, "externalCordapp")
    val cpbProjectDir = Files.createDirectory(testDir.resolve("cpb"))
```
and your `@Test` will still compile because it no longer uses these as `Path` properties.

Reply:
It is _trivial_ to demonstrate that none of `mavenRepoDir`, `externalCordappProjectDir` nor `cpbProjectDir` is used by the `@Test` function any more, but whatever. This code will change when JUnit 5.8 is released anyway.

Reply:
fixed, sorry... I didn't understand what you meant in the first place